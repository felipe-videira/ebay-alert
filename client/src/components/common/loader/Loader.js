import './Loader.scss';
import React from 'react'; 
import { Spin } from 'antd';

export const LOADER_SIZE = {
    SMALL: 'small',
    NORMAL: 'default',
    LARGE: 'large'
}

export default ({ 
    size = LOADER_SIZE.NORMAL, 
    visible = true
}) => (
    <Spin
        size={size} 
        spinning={visible} 
        className="loader"
    />
);