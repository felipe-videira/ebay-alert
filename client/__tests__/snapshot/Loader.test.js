import React from 'react';
import renderer from 'react-test-renderer';
import Loader, { LOADER_SIZE } from '../../src/components/common/loader/Loader';

describe('Loader snapshot tests', () => {
  it('renders correctly on default values', () => {
    expect(renderer.create(
      <Loader />
    ).toJSON())
    .toMatchSnapshot();
  });
  it('renders correctly on small size', () => {
    expect(renderer.create(
      <Loader size={LOADER_SIZE.SMALL} />
    ).toJSON())
    .toMatchSnapshot();
  });
  it('renders correctly on normal size', () => {
    expect(renderer.create(
      <Loader size={LOADER_SIZE.NORMAL}/>
    ).toJSON())
    .toMatchSnapshot();
  });
  it('renders correctly on large size', () => {
    expect(renderer.create(
      <Loader size={LOADER_SIZE.LARGE}/>
    ).toJSON())
    .toMatchSnapshot();
  });
  it('renders correctly on invisible mode', () => {
    expect(renderer.create(
      <Loader visible={false}/>
    ).toJSON())
    .toMatchSnapshot();
  });
  it('renders correctly on explicitly visible mode', () => {
    expect(renderer.create(
      <Loader visible />
    ).toJSON())
    .toMatchSnapshot();
  });
});

