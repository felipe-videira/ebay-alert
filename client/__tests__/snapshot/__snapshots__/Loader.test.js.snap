// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader snapshot tests renders correctly on default values 1`] = `
<div
  className="ant-spin ant-spin-spinning loader"
>
  <span
    className="ant-spin-dot ant-spin-dot-spin"
  >
    <i
      className="ant-spin-dot-item"
    />
    <i
      className="ant-spin-dot-item"
    />
    <i
      className="ant-spin-dot-item"
    />
    <i
      className="ant-spin-dot-item"
    />
  </span>
</div>
`;

exports[`Loader snapshot tests renders correctly on explicitly visible mode 1`] = `
<div
  className="ant-spin ant-spin-spinning loader"
>
  <span
    className="ant-spin-dot ant-spin-dot-spin"
  >
    <i
      className="ant-spin-dot-item"
    />
    <i
      className="ant-spin-dot-item"
    />
    <i
      className="ant-spin-dot-item"
    />
    <i
      className="ant-spin-dot-item"
    />
  </span>
</div>
`;

exports[`Loader snapshot tests renders correctly on invisible mode 1`] = `
<div
  className="ant-spin loader"
>
  <span
    className="ant-spin-dot ant-spin-dot-spin"
  >
    <i
      className="ant-spin-dot-item"
    />
    <i
      className="ant-spin-dot-item"
    />
    <i
      className="ant-spin-dot-item"
    />
    <i
      className="ant-spin-dot-item"
    />
  </span>
</div>
`;

exports[`Loader snapshot tests renders correctly on large size 1`] = `
<div
  className="ant-spin ant-spin-lg ant-spin-spinning loader"
>
  <span
    className="ant-spin-dot ant-spin-dot-spin"
  >
    <i
      className="ant-spin-dot-item"
    />
    <i
      className="ant-spin-dot-item"
    />
    <i
      className="ant-spin-dot-item"
    />
    <i
      className="ant-spin-dot-item"
    />
  </span>
</div>
`;

exports[`Loader snapshot tests renders correctly on normal size 1`] = `
<div
  className="ant-spin ant-spin-spinning loader"
>
  <span
    className="ant-spin-dot ant-spin-dot-spin"
  >
    <i
      className="ant-spin-dot-item"
    />
    <i
      className="ant-spin-dot-item"
    />
    <i
      className="ant-spin-dot-item"
    />
    <i
      className="ant-spin-dot-item"
    />
  </span>
</div>
`;

exports[`Loader snapshot tests renders correctly on small size 1`] = `
<div
  className="ant-spin ant-spin-sm ant-spin-spinning loader"
>
  <span
    className="ant-spin-dot ant-spin-dot-spin"
  >
    <i
      className="ant-spin-dot-item"
    />
    <i
      className="ant-spin-dot-item"
    />
    <i
      className="ant-spin-dot-item"
    />
    <i
      className="ant-spin-dot-item"
    />
  </span>
</div>
`;
